lib/covenants/namestate.js

```diff
@@ -740,96 +740,117 @@ class NameState extends bio.Struct {
+        const hours = ((blocks * spacing) / 60 / 60);
+
+        stats.openPeriodStart = start;
+        stats.openPeriodEnd = end;
+
+        stats.blocksUntilBidding = blocks;
+        stats.hoursUntilBidding = Number(hours.toFixed(2));
+        break;
+      }
+      case states.LOCKED: {
+        const start = this.height;
+        const end = this.height + lockupPeriod;
+        const blocks = end - height;
+        const hours = ((blocks * spacing) / 60 / 60);
+
+        stats.lockupPeriodStart = start;
+        stats.lockupPeriodEnd = end;
+
+        stats.blocksUntilClosed = blocks;
+        stats.hoursUntilClosed = Number(hours.toFixed(2));
+        break;
+      }
+      case states.BIDDING: {
+        const start = this.height + openPeriod;
+        const end = start + biddingPeriod;
+        const blocks = end - height;
+        const hours = ((blocks * spacing) / 60 / 60);
+
+        stats.bidPeriodStart = start;
+        stats.bidPeriodEnd = end;
+
+        stats.blocksUntilReveal = blocks;
+        stats.hoursUntilReveal = Number(hours.toFixed(2));
+        break;
+      }
+      case states.REVEAL: {
+        const start = this.height + openPeriod + biddingPeriod;
+        const end = start + revealPeriod;
+        const blocks = end - height;
+        const hours = ((blocks * spacing) / 60 / 60);
+
+        stats.revealPeriodStart = start;
+        stats.revealPeriodEnd = end;
+
+        stats.blocksUntilClose = blocks;
+        stats.hoursUntilClose = Number(hours.toFixed(2));
+        break;
+      }
+      case states.CLOSED: {
+        const start = this.renewal;
+        const end = start + renewalWindow;
+        const blocks = end - height;
+        const days = ((blocks * spacing) / 60 / 60 / 24);
+
+        stats.renewalPeriodStart = start;
+        stats.renewalPeriodEnd = end;
+
+        stats.blocksUntilExpire = blocks;
+        stats.daysUntilExpire = Number(days.toFixed(2));
+
+        // Add these details if name is in mid-transfer
+        if (this.transfer !== 0) {
+          const start = this.transfer;
+          const end = start + transferLockup;
+          const blocks = end - height;
+          const hours = ((blocks * spacing) / 60 / 60);
+
+          stats.transferLockupStart = start;
+          stats.transferLockupEnd = end;
+
+          stats.blocksUntilValidFinalize = blocks;
+          stats.hoursUntilValidFinalize = Number(hours.toFixed(2));
+        }
+        break;
+      }
+      case states.REVOKED: {
+        const start = this.revoked;
+        const end = start + auctionMaturity;
+        const blocks = end - height;
+        const hours = ((blocks * spacing) / 60 / 60);
+
+        stats.revokePeriodStart = start;
+        stats.revokePeriodEnd = end;
+
+        stats.blocksUntilReopen = blocks;
+        stats.hoursUntilReopen = Number(hours.toFixed(2));
+        break;
+      }
+      case EXPIRED: {
+        const expired = this.renewal + network.names.renewalWindow;
```

Comment:
renewalWindow is already imported and being used in line 811, perhaps don't need to reference `network.names.` here, otherwise the question is "are these constants different somehow?"

Reply:
You're absolutely right, thanks - actually its defined way up in L734, I'll push an update soon

Reply:
thank you! fixed and rebased